### PR TITLE
Left rotate array by d

### PR DESCRIPTION
**Pull request by RYzen-009**
Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

(Write your answer here.)
Yes

### Description
I have created a pull request for the issue  #5527 created on 11 April 2021
Left rotate array by d added to cp folder of the cpp file with proper explaination
Please review my pr

Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

#5560

### CODE REVIEW

File: C-Plus-Plus/cp/leftarraybyd.cpp [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+D=3
+4 5 6 7 8 9 10 1 2 3 //rotated array
+*/
+#include <bits/stdc++.h> 
+using namespace std; 
+void reversearray(int arr[],int start ,int end)
+{
+	
+	while (start<end){
+	swap(arr[start],arr[end]);             //swap function
```
Comment (bhav09) [inline C-Plus-Plus/cp/leftarraybyd.cpp]: Comments should be above the line of the code and not alongside.

File: C-Plus-Plus/cp/leftarraybyd.cpp [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+int main()
+{
+	int arr[] ={1,2,3,4,5,6,7};
+	int d=2;
+	int n = sizeof(arr)/sizeof(arr[0]); // comupting sizeof array
+	leftrotate(arr,d,n);
+	printarray(arr,n);
+return 0;	
+}
```
Comment (bhav09) [inline C-Plus-Plus/cp/leftarraybyd.cpp:49]: Mention the following things in this order:
1. Time complexity
2. Sample I/O

File: C-Plus-Plus/cp/leftarraybyd.cpp [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+There are many approach to slove this question like using the naive approach in which first perform
+left rotate by 1 code then left rotate function from 0 to d
+
+There is one more approach in which store d elements in temp then left rotate the remaining element
+then add the temp element back  arr[n-d+i]=temp[i]
+
+Now the efficient approach is using the Reversal Algorithm
+in this we use a function leftrotate and reverse function and we will be using reverse function  
+3 time 1st from o to d-1 then 2nd from  d to n-1 and then 3rd from o to n-1
+this will reverse in O(n) time
```
Comment (bhav09) [inline C-Plus-Plus/cp/leftarraybyd.cpp]: Remove complexity and Sample IO and mention them at EOF.

**Review by bhav09 — CHANGES_REQUESTED** [review]
Reviewed.

File: C-Plus-Plus/cp/leftarraybyd.cpp [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+1 2 3 4 5  
+D=2
+Output:- 3 4 5 1 2  
+
+INPUT:-
+n =10
+1 2 3 4 5 6 7 8 9 10 
+D=3
+Output :- 4 5 6 7 8 9 10 1 2 3 
+s*/
```
Comment (bhav09) [inline C-Plus-Plus/cp/leftarraybyd.cpp:66]: You need to add comments to the code. Remember the comments should be above the line segment of the code and not alongside. Also make a new line at EOF .

**Review by bhav09 — CHANGES_REQUESTED** [review]
Reviewed.

File: C-Plus-Plus/cp/leftarraybyd.cpp (hunk 4 above)
Comment (RYzen-009) [reply C-Plus-Plus/cp/leftarraybyd.cpp:66]: what do we you mean by make a new line at EOF

Comment (bhav09) [reply C-Plus-Plus/cp/leftarraybyd.cpp:66]: End Of FIle* :)

**Pull request closed without merging**